.github/workflows/release-new-version.yml

```diff
@@ -0,0 +1,96 @@
+name: Release New Version
+run-name: New release for "${{ github.ref_name }}" triggered by ${{ github.actor }}
+on:
+  pull_request:
+    types: [opened, synchronize]
+  # push:
+  #   branches:
+  #     - workflows
+jobs:
+  format_and_lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install Rust
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          components: rustfmt
+
+      - name: Rustfmt Check
+        uses: actions-rust-lang/rustfmt@v1
+
+      - name: Install Clippy
+        run: rustup update && rustup component add clippy
+
+      - name: Run Clippy
+        run: cargo clippy
+
+  end_to_end_tests:
+    runs-on: ubuntu-latest
+    needs: format_and_lint
+    environment: staging-fidl
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run tests
+        env:
+          GH_PRIVATE_KEY: ${{ secrets.GH_PRIVATE_KEY }}
+          DB_URL: ${{secrets.DB_URL}}
+        run: cargo test -- --nocapture
+
+  build_and_push:
+    runs-on: ubuntu-latest
+    needs: [format_and_lint, end_to_end_tests]
+    environment: production-fidl
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Cache Cargo registry
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+          key: ${{ runner.os }}-cargo-registry-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-registry-
+
+      - name: Cache Cargo build
+        uses: actions/cache@v2
+        with:
+          path: target
+          key: ${{ runner.os }}-cargo-build-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-build-
```

Comment:
Should use docker cache instead here.

Cargo cache should be used in 2 previous jobs instead.